frontend/app/features/password-reset/hooks/useResetPassword.ts

```diff
@@ -0,0 +1,56 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useResetPasswordMutation } from "@store/api/authSlice";
+import { useState } from "react";
+import { FieldErrors, useForm, UseFormHandleSubmit, UseFormRegister } from "react-hook-form";
+import * as z from "zod";
+
+const resetPasswordSchema = z.object({
+  email: z.string().email(),
+});
+
+type ResetPasswordFormData = z.infer<typeof resetPasswordSchema>;
+
+const useResetPassword = (): {
+  register: UseFormRegister<ResetPasswordFormData>;
+  handleSubmit: UseFormHandleSubmit<ResetPasswordFormData>;
+  resetPassword: (data: ResetPasswordFormData) => Promise<void>;
+  loading: boolean;
+  message: string;
+  errors: FieldErrors<ResetPasswordFormData>;
+} => {
+  const [message, setMessage] = useState("");
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<ResetPasswordFormData>({
+    resolver: zodResolver(resetPasswordSchema),
+  });
+
+  const [resetPasswordMutation, { isLoading }] = useResetPasswordMutation();
+
+  const resetPassword = async (data: ResetPasswordFormData) => {
+    setMessage("");
```

Comment:
please apply the same comment about try catch finally here.

Please always check all relative codes when you see a comment on the PR, the comment is not only relates to the specific code, most comments are general, so you always have to check your entire changes to make sure there is no related issue, 
it also saves time for both me and you, its really disappointing to comment the same issue over again